trio/_core/_io_windows.py

```diff
@@ -185,13 +188,15 @@ class CKeys(enum.IntEnum):
     USER_DEFINED = 4  # and above
 
 
-def _check(success):
+def _check(success: bool) -> Literal[True]:
```

Comment:
At some point I figured out that this needs to be a TypeVar, it's used on return values to raise when they indicate errors.